src/main/java/com/revature/assignforce/service/BatchServiceImpl.java

```diff
@@ -1,46 +1,99 @@
-package com.revature.assignforce.service;
-
-import java.util.List;
-import java.util.Optional;
-
-import javax.transaction.Transactional;
-
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Service;
-
-import com.revature.assignforce.beans.Batch;
-import com.revature.assignforce.repos.BatchRepository;
-
-@Transactional
-@Service
-public class BatchServiceImpl implements BatchService {
-
-	@Autowired
-	private BatchRepository batchRepository;
-	
-	@Override
-	public List<Batch> getAll() {
-		return batchRepository.findAll();
-	}
-
-	@Override
-	public Optional<Batch> findById(int id) {
-		return batchRepository.findById(id);
-	}
-
-	@Override
-	public Batch update(Batch b) {
-		return batchRepository.save(b);
-	}
-
-	@Override
-	public Batch create(Batch b) {
-		return batchRepository.save(b);
-	}
-
-	@Override
-	public void delete(int id) {
-		batchRepository.deleteById(id);
-	}
-
-}
+package com.revature.assignforce.service;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.transaction.Transactional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.revature.assignforce.beans.Batch;
+import com.revature.assignforce.beans.SkillIdHolder;
+import com.revature.assignforce.commands.FindCurriculumCommand;
+import com.revature.assignforce.commands.FindLocationCommand;
+import com.revature.assignforce.commands.FindSkillsCommand;
+import com.revature.assignforce.commands.FindTrainerCommand;
+import com.revature.assignforce.repos.BatchRepository;
+import com.revature.assignforce.repos.SkillRepository;
+
+@Transactional
+@Service
+public class BatchServiceImpl implements BatchService {

```

Comment:
As the level complexity of some methods in this class has been increased, it would be useful to add some documentation comments to describe what they are doing and extract any behavior that can be isolated to private methods.